pkg/cloudprovider/aws/fleet/factory.go

```diff
@@ -0,0 +1,75 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fleet
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/iam/iamiface"
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"github.com/patrickmn/go-cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	CacheTTL              = 5 * time.Minute
+	CacheCleanupInterval  = 10 * time.Minute
```

Comment:
Would be nice to have a comment explaining these (or a TODO if they're still a WIP?)

Reply:
I actually think these are a fine baseline, for now. Results in one request per cache per 5 minutes. Updates to new user configs within 5 minutes. I could see going as low as 1 minute, though. WDYT?

Reply:
Sounds fine - could you just add a comment ... sometimes 10 years later people are wondering if this kind of thing was the result of a lot of experimentation.